spec/acts_as_tenant/model_extensions_spec.rb

```diff
@@ -199,14 +199,46 @@
   end
 
   describe "It should be possible to use associations with foreign_key from polymorphic" do
-    before do
-      @account = Account.create!(name: 'foo')
-      ActsAsTenant.current_tenant = @account
-      @project = Project.create!(name: 'project', account: @account)
-      @comment = Comment.new commentable: @project, account: @account
+    context 'tenanted objects have a polymorphic association' do
+      before do
+        @account = Account.create!(name: 'foo')
+        ActsAsTenant.current_tenant = @account
+        @project = Project.create!(name: 'project', account: @account)
+        @comment = Comment.new commentable: @project, account: @account
+      end
+
+      it { expect(@comment.save!).to eq(true) }
     end
 
-    it { expect(@comment.save!).to eq(true) }
+    context 'tenant is polymorphic' do
+      before do
+        @account = Account.create!(name: 'foo')
+        @project = Project.create!(name: 'polymorphic project')
+        ActsAsTenant.current_tenant = @project
+        @comment = PolymorphicTenantComment.new(account: @account)
+      end
+
+      it 'populates commentable_type with the current tenant' do
+        expect(@comment.polymorphic_tenant_commentable_id).to eql(@project.id)
+        expect(@comment.polymorphic_tenant_commentable_type).to eql(@project.class.to_s)
+      end
+
+      context 'with another type of tenant, same id' do
+        before do
+          @comment.save!
+          @article = Article.create!(id: @project.id, title: 'article title')
+          @comment_on_article = @article.polymorphic_tenant_comments.create!
+        end
+
+        it 'correctly scopes to the current tenant type' do
+          expect(@comment_on_article).to be_persisted
+          expect(@comment).to be_persisted
+          expect(PolymorphicTenantComment.count).to eql(1)
+          expect(PolymorphicTenantComment.all.first.attributes).to eql(@comment.attributes)
+        end

```

Comment:
Scopes needed to be updated as well or else querying for all the comments would pull comments under both Article and Project for the current tenant ID number. This PR introduces type checking of the tenant into the scope when the `polymorphic` option is passed.

Reply:
Where are you updating those scopes. Are you referring to your original PR?